src/atoms/visual-story/amp-story-auto-ads/amp-story-auto-ads.tsx

```diff
@@ -17,36 +17,31 @@ export const AmpStoryAutoAdsBase = ({ story, config }: AutoAdsTypes) => {
           src="https://cdn.ampproject.org/v0/amp-story-auto-ads-0.1.js"
         />
       </Helmet>
-      <amp-story-auto-ads>
-        {doubleClick && (
-          <script
-            type="application/json"
-            dangerouslySetInnerHTML={{
-              __html: JSON.stringify({
-                "ad-attributes": {
-                  type: "doubleclick",
-                  "data-slot": doubleClick,
-                  json: getTargetingInfo({ config, story })
-                }
-              })
-            }}
-          />
-        )}
-        {adsense.clientId && (
-          <script
-            type="application/json"
-            dangerouslySetInnerHTML={{
-              __html: JSON.stringify({
-                "ad-attributes": {
-                  type: "adsense",
-                  "data-ad-client": adsense.clientId,
-                  "data-ad-slot": adsense.slotId
-                }
-              })
-            }}
-          />
-        )}
-      </amp-story-auto-ads>
+      {
+        (doubleClick || adsense.clientId || mgId.widgetId) && (
+          <amp-story-auto-ads>
+            <script
+              type="application/json"
+              dangerouslySetInnerHTML={{
+                __html: JSON.stringify({
+                  "ad-attributes": {
+                    type: doubleClick ? "doubleclick" : adsense.clientId ? "adsense" : "mgid",
+                    ...(doubleClick && {
+                      "data-slot": doubleClick,
+                      json: getTargetingInfo({ config, story })
+                    }),
+                    ...(adsense.clientId && {
+                      "data-ad-client": adsense.clientId,
+                      "data-ad-slot": adsense.slotId
+                    }),
+                    ...(mgId.widgetId && { "data-widget": mgId.widgetId })
```

Comment:
I'd suggest constructing the config object above the return statement and not add it in jsx. Also would suggest creating 3 config objects (one per ad provider) instead of just one config obj to reduce chances of errors and make it more readable

Reply:
Changed the architecture